### PR TITLE
Improve error messages and validation for admin pages

### PR DESCRIPTION
**Pull request by jacobherrington**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [ ] Feature
- [x] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description

We should be rending error messages here and this form needs some client-side validation.

## Related Tickets & Documents

Closes https://github.com/forem/InternalProjectPlanning/issues/232

## Added tests?

- [x] Yes
- [ ] No, and this is why: _please replace this line with details on why tests
      have not been included_
- [ ] I need help with writing tests

## Added to documentation?

- [ ] Docs.forem.com
- [ ] README
- [x] No documentation needed

### CODE REVIEW

File: spec/system/admin/admin_manages_pages_spec.rb [hunk 1]
```diff
@@ -54,6 +54,18 @@
       end
     end
 
+    it "has client-side validation" do
```
Comment (jacobherrington) [inline spec/system/admin/admin_manages_pages_spec.rb:57]: This is not the sturdiest of all specs, but testing for HTML validation is a bit tricky, and if this fails we know something changed, which will help find any bugs.

File: app/controllers/admin/pages_controller.rb [hunk 2]
```diff
@@ -23,14 +23,25 @@ def edit
...
     def update
       @page = Page.find(params[:id])
-      @page.update!(page_params)
-      redirect_to "/admin/pages"
+      @page.assign_attributes(page_params)
+      if @page.valid?
+        @page.update!(page_params)
+        redirect_to admin_pages_path
+      else
+        flash.now[:error] = @page.errors.full_messages.join(", ")
```
Comment (Zhao-Andy) [inline app/controllers/admin/pages_controller.rb:31]: I like using `model.errors_as_sentence` :)

https://github.com/forem/forem/blob/531f26bad69e6cb03468d1f84d2e4a64c3dea757/app/models/application_record.rb#L56-L58

Comment (Zhao-Andy) [reply app/controllers/admin/pages_controller.rb:31]: Also curious, why `flash.now` over regular `flash`? So that it doesn't have a chance to persist?

**Review by Zhao-Andy — APPROVED** [review]
LGTM! Couple of comments, but non blocking.

**Review by juliannatetreault — APPROVED** [review]
Nice improvements! 👍

File: app/controllers/admin/pages_controller.rb (hunk 2 above)
Comment (jacobherrington) [reply app/controllers/admin/pages_controller.rb]: Cool, I'll do that! Thanks!

`flash` uses the flash message on the next request (so would be ideal before `redirect_to`), whereas `flash.now` will use the flash message on the current request (so generally, called before `render`).

**Pull request merged**